core/Translation/Transifex/API.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Translation\Transifex;
+
+use Exception;
+use Piwik\Exception\AuthenticationFailedException;
+
+class API
+{
+    protected $apiUrl = 'https://www.transifex.com/api/2/';
+    protected $username = '';
+    protected $password = '';
+    protected $projectSlug = '';
+
+    public function __construct($username, $password, $project='piwik')
+    {
+        $this->username = $username;
+        $this->password = $password;
+        $this->projectSlug = $project;
+    }
+
+    /**
+     * Returns all resources available on Transifex project
+     *
+     * @return array
+     */
+    public function getAvailableResources()
+    {
+        static $resources;
```

Comment:
OK I see, but those resources depend on the state of the class. If I create a new instance of the class with a different Transifex project this is broken. I agree that this might never happen though, but using the cache system might be a better option as it's what we want to do (cache the results)? And probably not so much complicated (I don't really know I've never used it yet)?